README.md

```diff
@@ -850,6 +855,54 @@ class MyComponent(Component):
             do_something_extra(request, *args, **kwargs)
 ```
 
+## Pre-defined components
+
+### Dynamic components
+
+If you are writing something like a form component, you may design it such that users
+give you the component names, and your component renders it.
+
+While you can handle this with a series of if / else statements, this is not an extensible solution.
+
+Instead, you can use **dynamic components**. Dynamic components are used in place of normal components.
+
+```django
+{% load component_tags %}
+{% component "dynamic" is=component_name title="Cat Museum" %}
+    {% fill "content" %}
+        HELLO_FROM_SLOT_1
+    {% endfill %}
+    {% fill "sidebar" %}
+        HELLO_FROM_SLOT_2
+    {% endfill %}
+{% endcomponent %}
+```
+
+These behave same way as regular components. You pass it the same args, kwargs, and slots as you would
+to the component that you want to render.
+
+The only exception is that also you supply 1-2 additional inputs:
+- `is` - Required - The component name or a component class to render
+- `registry` - Optional - The `ComponentRegistry` that will be searched if `is` is a component name. If omitted, ALL registries are searched.
+
+By default, the dynamic component is registered under the name `"dynamic"`. In case of a conflict, you can change the name used for the dynamic components by defining the [`COMPONENTS.dynamic_component_name` setting](#dynamic_component_name).
```

Comment:
Should we add a test for what happens on conflicts?

Reply:
Good idea! Added in https://github.com/EmilStenstrom/django-components/pull/627/commits/2bd105b03837efc4332704cf649b6b332d1a0e28